src/System.Data.Common/src/System/Data/Common/IDbColumnSchemaGenerator.cs

```diff
@@ -0,0 +1,11 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Data.Common
+{
+    public interface IDbColumnSchemaGenerator
```

Comment:
Ah, too bad about the conflicting concept... Though, I still think "Generator" doesn't quite fit: the schema isn't being generated (i.e. created), right? It's being returned (i.e. provided) by the reader.

How about `IDbColumnSchemaProducer`? I'll admit "Producer" isn't perfect: it still has a create/generate connotation, but "produce" also means "to provide, furnish, or supply; yield" which is a little closer to what this interface is doing.

Any other naming ideas?

Some others I thought of:
 - `IDbColumnSchemaReader`
 - `IDbColumnSchemaRetriever`
 - `IDbColumnSchemaGetter`
 - `IDbColumnSchemaObtainer`
 - `IDbColumnSchemaFetcher`
 - `IDbColumnSchemaSupplier`